src/array/list/mod.rs

```diff
@@ -54,7 +54,7 @@ impl<O: Offset> ListArray<O> {
 
         let child_data_type = Self::try_get_child(&data_type)?.data_type();
         let values_data_type = values.data_type();
-        if child_data_type != values_data_type {
+        if child_data_type.to_logical_type() != values_data_type.to_logical_type() {
```

Comment:
don't we want this to remain? I.e. if the datatype is `List<Extension(Int32)>` but the array is `List<Int32>`, we want to error, right?

Reply:
Yep, I missed the actual root-cause of the problem, which was in `GrowableStruct`.